packages/react-data-grid/src/ColumnComparer.ts

```diff
@@ -0,0 +1,24 @@
+import { isElement } from 'react-is';
+
+export function sameColumn<A, B>(a: A, b: B): boolean {
+  for (const k in a) {
+    if (a.hasOwnProperty(k)) {
+      const valA = a[k] as unknown;
+      const valB = b[k as string as keyof B] as unknown;
```

Comment:
Without the `as string as keyof B` part:
![image](https://user-images.githubusercontent.com/567105/55960077-f5fc9680-5c63-11e9-870a-b1016d16c22a.png)

Without at least one `as unknown`:
![image](https://user-images.githubusercontent.com/567105/55960200-30663380-5c64-11e9-8e83-08637ba7f965.png)